### PR TITLE
Issue #125 - Addressing missing server url in check_signature method

### PR DESCRIPTION
**Pull request by rbebersole**
The verify_discovery_results method call was moved after the check_signature method call to address a security vulnerability (issue #121).  Apparently, the check_signature method relied on the endpoint instance variable being defined by the verify_discovery_results method.   As a result, the check_signature method always fails because the server_url is nil.

This fix initializes the endpoint variable within the check_signature method and populates it with the server url passed by the OPENID2 client.  

Note:  I followed the link for instructions on contributing, but I could not find any instructions on the target site.  So I used what the other contributors did as a guide.

### CODE REVIEW

**Comment by rbebersole** [comment]
I believe the error in the failed check is related to the installing rubinius-3 and not from the coding change.

File: lib/openid/consumer/idres.rb [hunk 1]
```diff
@@ -203,6 +203,21 @@ def verify_return_to_base(msg_return_to)
 
       # Raises ProtocolError if the signature is bad
       def check_signature
+        # ----------------------------------------------------------------------
+        # The server url must be defined within the endpoint instance for the
+        # OpenID2 namespace in order for the signature check to complete
+        # successfully.
+        #
+        # This fix corrects issue #125 - Unable to complete OpenID login
+        #                                with ruby-openid 2.9.0/2.9.1
```
Comment (utkarsh2102) [inline lib/openid/consumer/idres.rb:212]: Probably we can ✂️  this out from here?

**Review by utkarsh2102 — COMMENTED** [review]
This looks good, thanks for the PR!
@tobiashm, could you take a final look and merge and release?

File: lib/openid/consumer/idres.rb (hunk 1 above)
Comment (rbebersole) [reply lib/openid/consumer/idres.rb:212]: That is fine to remove those two comment lines during the merge.

**Comment by timcappalli** [comment]
This repo is being archived. Closing PR.

**Pull request closed without merging**